### ISSUE TITLE
Error Docker Business Central 3221226505

### ISSUE THREAD

**Issue by luisignaciog** [microsoft/navcontainerhelper#2505]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
When I try to create a Business Central container, at step 5/6 after executing the command "Copying ReportBuilder" it gives the following error:
The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; \Run\start.ps1 -installOnly -includeTestToolkit -includePerformanceToolkit' returned a non-zero code: 3221226505

**Scripts used to create container and cause the issue**
```
$artifactUrl = Get-BCArtifactUrl -type sandbox -country es -select Latest
$containername = 'BC005'
$licenseFile = [URL with license file]
$credential = New-Object pscredential [user], (ConvertTo-SecureString -String [password] -AsPlainText -Force)

New-BcContainer `
    -accept_eula `
    -useSSL:$false `
    -containerName $containerName `
    -credential $credential `
    -auth NavUserPassword `
    -artifactUrl $artifactUrl `
    -imageName $containerName `	
    -includeTestToolkit -includeTestLibrariesOnly `
    -multitenant:$false `
    -assignPremiumPlan `
    -licenseFile $licenseFile `
	-Credential $Credential `
    -dns '8.8.8.8' `
    -memoryLimit 4G `	
    -updateHosts
```

**Full output of scripts**
```
BcContainerHelper version 3.0.7
BcContainerHelper emits usage statistics telemetry to Microsoft
BcContainerHelper is version 3.0.7
BcContainerHelper is running as administrator
Hyper-V is Enabled
Host is Microsoft Windows 11 Pro - 21H2
Docker Client Version is 20.10.14
Docker Server Version is 20.10.14
Downloading application artifact /sandbox/20.1.39764.41057/es
Downloading C:\Users\username\AppData\Local\Temp\7fbb9e46-e120-4f87-b893-489672210222.zip
Unpacking application artifact to tmp folder using 7zip
Downloading platform artifact /sandbox/20.1.39764.41057/platform
Downloading C:\Users\username\AppData\Local\Temp\54d2acc7-d933-451b-b98b-e3c7e74d2bba.zip
Unpacking platform artifact to tmp folder using 7zip
Downloading Prerequisite Components
Downloading c:\bcartifacts.cache\sandbox\20.1.39764.41057\platform\Prerequisite Components\Open XML SDK 2.5 for Microsoft Office\OpenXMLSDKv25.msi
Downloading c:\bcartifacts.cache\sandbox\20.1.39764.41057\platform\Prerequisite Components\IIS URL Rewrite Module\rewrite_2.0_rtw_x64.msi
Downloading c:\bcartifacts.cache\sandbox\20.1.39764.41057\platform\Prerequisite Components\DotNetCore\DotNetCore.1.0.4_1.1.1-WindowsHosting.exe
Fetching all docker images
Fetching all docker volumes
ArtifactUrl and ImageName specified
WARNING: Unable to find matching generic image for your host OS. Using mcr.microsoft.com/businesscentral:10.0.20348.707
Building multitenant image bc005:sandbox-20.1.39764.41057-es-mt based on mcr.microsoft.com/businesscentral:10.0.20348.707 with https://bcartifacts.azureedge.net/sandbox/20.1.39764.41057/es
Pulling latest image mcr.microsoft.com/businesscentral:10.0.20348.707
10.0.20348.707: Pulling from businesscentral
97f65a0ec59e: Already exists
12fb50a031bd: Already exists
37213f33d2a7: Already exists
8a2eb072c814: Already exists
bd41621c0ab5: Already exists
eff7da49bebe: Already exists
491c44f093ad: Already exists
0261770c003c: Already exists
a006dc2c1436: Already exists
1d1985dcf9d4: Already exists
5fa0873c1b9c: Already exists
08b6141a7569: Already exists
26f97c07b305: Already exists
65dfe50c63d2: Already exists
c5bbf01420d3: Already exists
61dbb9d15985: Pulling fs layer
2bd9b24600b4: Pulling fs layer
61dbb9d15985: Verifying Checksum
61dbb9d15985: Download complete
61dbb9d15985: Pull complete
2bd9b24600b4: Pull complete
Digest: sha256:ff37f262dec05d68de5e3169421f6e2d8cd020eaf47fbe0abf7b15af55661595
Status: Downloaded newer image for mcr.microsoft.com/businesscentral:10.0.20348.707
mcr.microsoft.com/businesscentral:10.0.20348.707
Generic Tag: 1.0.2.7
Container OS Version: 10.0.20348.707 (ltsc2022)
Host OS Version: 10.0.22000.675 (21H2)
Using process isolation
Files in c:\bcartifacts.cache\1ywfwbpt.h0c\my:
Copying Platform Artifacts
c:\bcartifacts.cache\sandbox\20.1.39764.41057\platform
Copying Database
Copying Licensefile
Copying ConfigurationPackages
C:\bcartifacts.cache\sandbox\20.1.39764.41057\es\ConfigurationPackages
Copying Extensions
C:\bcartifacts.cache\sandbox\20.1.39764.41057\es\Extensions
Copying Applications.ES
C:\bcartifacts.cache\sandbox\20.1.39764.41057\es\Applications.ES
c:\bcartifacts.cache\1ywfwbpt.h0c
Sending build context to Docker daemon  2.217GB

Step 1/6 : FROM mcr.microsoft.com/businesscentral:10.0.20348.707
 ---> bb3136058621
Step 2/6 : ENV DatabaseServer=localhost DatabaseInstance=SQLEXPRESS DatabaseName=CRONUS IsBcSandbox=Y artifactUrl=https://bcartifacts.azureedge.net/sandbox/20.1.39764.41057/es filesOnly=False
 ---> Running in 90e203449b79
Removing intermediate container 90e203449b79
 ---> 1893fb996adc
Step 3/6 : COPY my /run/
 ---> 9f659dff4417
Step 4/6 : COPY NAVDVD /NAVDVD/
 ---> f985da9caf00
Step 5/6 : RUN \Run\start.ps1 -installOnly -multitenant
 ---> Running in 6a9cd376f04c
c:\run\my folder doesn't exist, creating it
Using installer from C:\Run\150-new
Installing Business Central                                                                                             Installing from DVD                                                                                                     Starting Local SQL Server                                                                                               Starting Internet Information Server                                                                                    Copying Service Tier Files
C:\NAVDVD\ServiceTier\Program Files
C:\NAVDVD\ServiceTier\System64Folder
Copying PowerShell Scripts
C:\NAVDVD\WindowsPowerShellScripts\Cloud\NAVAdministration
C:\NAVDVD\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
C:\NAVDVD\WebClient\Microsoft Dynamics NAV
Copying Client Files
C:\NAVDVD\LegacyDlls\program files\Microsoft Dynamics NAV
C:\NAVDVD\LegacyDlls\program files\Microsoft Dynamics NAV
C:\NAVDVD\LegacyDlls\systemFolder
Copying ModernDev Files
C:\NAVDVD
C:\NAVDVD\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\NAVDVD\ConfigurationPackages
Copying Test Assemblies
C:\NAVDVD\Test Assemblies
Copying Extensions
C:\NAVDVD\Extensions
Copying Applications
C:\NAVDVD\Applications
Copying Applications.ES
C:\NAVDVD\Applications.ES
Copying dependencies
Copying ReportBuilder
The command 'powershell -Command $ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue'; \Run\start.ps1 -installOnly -multitenant' returned a non-zero code: 3221226505
New-Bcimage Telemetry Correlation Id: 1c3d37b9-4207-49aa-adb7-dca89d01fed3
New-BcContainer Telemetry Correlation Id: d04dc505-a4d8-4b0e-8155-c66930c1e233
Docker Build failed with exit code -1073740791
En C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\3.0.7\ContainerHandling\New-NavImage.ps1: 637
Carácter: 25
+ ...              throw "Docker Build failed with exit code $LastExitCode"
+                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Docker Build fa...ode -1073740791:String) [], RuntimeException
    + FullyQualifiedErrorId : Docker Build failed with exit code -1073740791
```

**Additional context**
I have tried with the wizard and the same

**Comment by freddydk** [comment]
Please try to remove the memorylimit and the imagename to see whether this is related to lack of memory or image building.

**Comment by luisignaciog** [comment]
Hi, same error with the memorylimit removed

$artifactUrl = Get-BCArtifactUrl -type sandbox -country es -select Latest
$containername = 'BC005'
$licenseFile = [URL with license file]
$credential = New-Object pscredential [user], (ConvertTo-SecureString -String [password] -AsPlainText -Force)

New-BcContainer `
    -accept_eula `
    -useSSL:$false `
    -containerName $containerName `
    -credential $credential `
    -auth NavUserPassword `
    -artifactUrl $artifactUrl `
    -imageName $containerName `	
    -includeTestToolkit -includeTestLibrariesOnly `
    -multitenant:$false `
    -assignPremiumPlan `
    -licenseFile $licenseFile `
	-Credential $Credential `
    -dns '8.8.8.8' `    
    -updateHosts

**Comment by luisignaciog** [comment]
Hi, if I only remove -imagename the container is created perfectly but I'cant acces to web client.

The hosts file I see it well, I will continue to investigate what it can be, but the container has been created. Thank you

**Comment by freddydk** [comment]
what happens when you try navigating to the web client?
Does it change anything if you run Restart-BcContainer after you setup the container?

**Comment by luisignaciog** [comment]
I didn't change anything but I restart the container and now it works!! Thanks!!!

**Comment by freddydk** [comment]
Next version of containerhelper will perform a healthcheck after creating the container.
If this fails, it will attempt a restart (once) to see whether this fixes things.
This should mitigate this issue.
If you start the container with -doNotCheckHealth - the check is skipped.

**Comment by freddydk** [comment]
Shipped in BcContainerHelper 3.0.11
